core/jobs/transforms/job_result_transforms.py

```diff
@@ -0,0 +1,149 @@
+
+        Args:
+            prefix: str|None. The prefix for the result string.
+            label: str|None. The label of the PTransform.
+        """
+        super().__init__(label=label)
+        self.prefix = '%s ' % prefix if prefix else ''
+
+    # This is needed because the Beam annotations validator doesn't properly
+    # work with result.Result.
+    @beam.typehints.no_annotations
+    def _transform_result_to_job_run_result(
+        self, result_item: result.Result[Any, Any]
+    ) -> job_run_result.JobRunResult:
+        """Transforms Result objects into JobRunResult objects. When the result
+        is Ok then transform it into stdout, if the result is Err transform it
+        into stderr.
+
+        Args:
+            result_item: Result. The result object.
+
+        Returns:
+            JobRunResult. The JobRunResult object.
+        """
+        if isinstance(result_item, result.Ok):
+            return job_run_result.JobRunResult.as_stdout(
+                '%sSUCCESS:' % self.prefix
+            )
+        else:
+            return job_run_result.JobRunResult.as_stderr(
+                '%sERROR: "%s":' % (self.prefix, result_item.value)
+            )
+
+    @staticmethod
+    def _add_count_to_job_run_result(
+        job_result_and_count: Tuple[job_run_result.JobRunResult, int]
+    ) -> job_run_result.JobRunResult:
+        """Adds count to the stdout or stderr of the JobRunResult.
+
+        Args:
+            job_result_and_count: tuple(JobRunResult, int). Tuple containing
+                unique JobRunResult and their counts.
+
+        Returns:
+            JobRunResult. JobRunResult objects with counts added.
+        """
+        job_result, count = job_result_and_count
+        if job_result.stdout:
+            job_result.stdout += ' %s' % str(count)
+        if job_result.stderr:
+            job_result.stderr += ' %s' % str(count)
+        return job_result
+
+    # This is needed because the Beam annotations validator doesn't properly
+    # work with result.Result.
+    @beam.typehints.no_annotations
+    def expand(
+        self, results: beam.PCollection[result.Result[Any, Any]]
+    ) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Transforms Result objects into unique JobRunResult objects and
+        adds counts to them.
+
+        Args:
+            results: PCollection. Sequence of Result objects.
+
+        Returns:
+            PCollection. Sequence of unique JobRunResult objects with count.
+        """
+        return (
+            results
+            | 'Transform result to job run result' >> beam.Map(
+                self._transform_result_to_job_run_result)
+            | 'Count all elements' >> beam.combiners.Count.PerElement()
+            | 'Add count to job run result' >> beam.Map(
+                self._add_count_to_job_run_result)
+        )
+
+
+class CountObjectsToJobRunResult(beam.PTransform):
+    """Count sequence of any object into job_run_result.JobRunResult."""
```

Comment:
I don't understand what this means. Can you elaborate (perhaps with an example)? What do you mean by "count X into Y"? (That isn't really standard English.)

You might need to use more words to describe the idea that you're trying to convey.

Reply:
Done, rephrased.